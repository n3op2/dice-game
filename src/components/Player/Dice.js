import React from 'react';

import up from '/Users/paulius/dice-game/assets/dice-one-solid.svg';
import back from '/Users/paulius/dice-game/assets/dice-two-solid.svg';
import left from '/Users/paulius/dice-game/assets/dice-three-solid.svg';
import right from '/Users/paulius/dice-game/assets/dice-four-solid.svg';
import front from '/Users/paulius/dice-game/assets/dice-five-solid.svg';
import down from '/Users/paulius/dice-game/assets/dice-six-solid.svg';

const sides = {
    1: { name: 'top', img: up },
    2: { name: 'back', img: back },
    3: { name: 'left', img: left },
    4: { name: 'right', img: right },
    5: { name: 'front', img: front },
    6: { name: 'bottom', img: down },
};

const Dice = ({ rolling, val }) => {
    const [side, setSide] = React.useState({ img: up, name: 'up' });

    React.useEffect(() => {
        if (val === 0) return null;
        setSide(sides[val]);
    }, [rolling, val]);

    return (
        <img
            name={side.name}
            className={rolling ? 'dice-rolling' : 'dice'}
            src={side.img}
        />
    );
};

export default Dice;
