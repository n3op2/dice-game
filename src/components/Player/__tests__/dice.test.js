import React from 'react';
import { mount } from 'enzyme';
import Dice from '../Dice';

const diceValues = [
    { val: 1, name: 'top' },
    { val: 2, name: 'back' },
    { val: 3, name: 'left' },
    { val: 4, name: 'right' },
    { val: 5, name: 'front' },
    { val: 6, name: 'bottom' },
];

describe('Dice Component', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(<Dice rolling={true} val={2} />);
    });

    test('renders a component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('sets img class to [dice-rolling] when rolling is in action', () => {
        wrapper.setProps({ rolling: true });
        expect(wrapper.find('img').prop('className')).toBe('dice-rolling');
    });

    test('sets img class to dice after rolling', () => {
        wrapper.setProps({ rolling: false });
        expect(wrapper.find('img').prop('className')).toBe('dice');
    });

    describe('mapping values to dice sides', () => {
        diceValues.forEach((dice) => {
            test(`for [${dice.val}] value renders -> ${dice.name} side`, () => {
                wrapper.setProps({ val: dice.val });
                wrapper.update();
                expect(wrapper.find('img')).toMatchSnapshot();
                expect(wrapper).toMatchSnapshot();
            });
        });
    });
});
