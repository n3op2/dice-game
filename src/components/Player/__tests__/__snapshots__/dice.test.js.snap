// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dice Component mapping values to dice sides for [1] value renders -> top side 1`] = `
<img
  className="dice-rolling"
  name="top"
  src={Object {}}
/>
`;

exports[`Dice Component mapping values to dice sides for [1] value renders -> top side 2`] = `
<Dice
  rolling={true}
  val={1}
>
  <img
    className="dice-rolling"
    name="top"
    src={Object {}}
  />
</Dice>
`;

exports[`Dice Component mapping values to dice sides for [2] value renders -> back side 1`] = `
<img
  className="dice-rolling"
  name="back"
  src={Object {}}
/>
`;

exports[`Dice Component mapping values to dice sides for [2] value renders -> back side 2`] = `
<Dice
  rolling={true}
  val={2}
>
  <img
    className="dice-rolling"
    name="back"
    src={Object {}}
  />
</Dice>
`;

exports[`Dice Component mapping values to dice sides for [3] value renders -> left side 1`] = `
<img
  className="dice-rolling"
  name="left"
  src={Object {}}
/>
`;

exports[`Dice Component mapping values to dice sides for [3] value renders -> left side 2`] = `
<Dice
  rolling={true}
  val={3}
>
  <img
    className="dice-rolling"
    name="left"
    src={Object {}}
  />
</Dice>
`;

exports[`Dice Component mapping values to dice sides for [4] value renders -> right side 1`] = `
<img
  className="dice-rolling"
  name="right"
  src={Object {}}
/>
`;

exports[`Dice Component mapping values to dice sides for [4] value renders -> right side 2`] = `
<Dice
  rolling={true}
  val={4}
>
  <img
    className="dice-rolling"
    name="right"
    src={Object {}}
  />
</Dice>
`;

exports[`Dice Component mapping values to dice sides for [5] value renders -> front side 1`] = `
<img
  className="dice-rolling"
  name="front"
  src={Object {}}
/>
`;

exports[`Dice Component mapping values to dice sides for [5] value renders -> front side 2`] = `
<Dice
  rolling={true}
  val={5}
>
  <img
    className="dice-rolling"
    name="front"
    src={Object {}}
  />
</Dice>
`;

exports[`Dice Component mapping values to dice sides for [6] value renders -> bottom side 1`] = `
<img
  className="dice-rolling"
  name="bottom"
  src={Object {}}
/>
`;

exports[`Dice Component mapping values to dice sides for [6] value renders -> bottom side 2`] = `
<Dice
  rolling={true}
  val={6}
>
  <img
    className="dice-rolling"
    name="bottom"
    src={Object {}}
  />
</Dice>
`;

exports[`Dice Component renders a component 1`] = `
<Dice
  rolling={true}
  val={2}
>
  <img
    className="dice-rolling"
    name="back"
    src={Object {}}
  />
</Dice>
`;
